File: Server/src/main/java/com/springboot/garageApp/service/CustomerService.java
package com.springboot.garageApp.service;

import com.springboot.garageApp.model.Customer;
import com.springboot.garageApp.repository.CustomerRepository;
import org.springframework.stereotype.Service;

import java.util.ArrayList;
import java.util.List;

@Service
public class CustomerService {

    private CustomerRepository customerRepository;

    public CustomerService(CustomerRepository customerRepository) {
        this.customerRepository = customerRepository;
    }

    public List<Customer> getCustomers(){
        List<Customer> customerList = new ArrayList<>();
        customerRepository.findAll().forEach(customer ->
                customerList.add(customer)
        );
        return customerList;
    }

    public Customer getCustomer(long id){
        return customerRepository.findById(id).orElse(new Customer());
    }

    public void addCustomer(Customer customer){
        customerRepository.save(customer);
    }

    public void deleteCustomer(long id){
        customerRepository.deleteById(id);
    }

    public void updateCustomer(Customer customer, long id){
        Customer customerInDB = customerRepository.findById(id).get();
        if(customerInDB != null) {
            if(customer.getLastName() != null)
                customerInDB.setLastName(customer.getLastName());
            if(customer.getFirstName() != null)
                customerInDB.setFirstName(customer.getFirstName());
            if(customer.getMail() != null)
                customerInDB.setMail(customer.getMail());
            if(customer.getAddress() != null)
                customerInDB.setAddress(customer.getAddress());
            if(customer.getPhoneNumber() != 0)
                customerInDB.setPhoneNumber(customer.getPhoneNumber());
            if(customer.getAge() != 0)
                customerInDB.setAge(customer.getAge());
            customerRepository.save(customerInDB);
        }
    }
}

File: Server/src/main/java/com/example/garageApp/service/CustomerService.kt
package com.example.garageApp.service

import com.example.garageApp.model.Customer
import com.example.garageApp.repository.CustomerRepository
import org.springframework.stereotype.Service
import java.util.function.Consumer

@Service
class CustomerService(private val customerRepository: CustomerRepository) {
    val customers: List<Customer>
        get() {
            val customerList: MutableList<Customer> = ArrayList()
            customerRepository.findAll().forEach(
                Consumer { customer: Customer? ->
                    if (customer != null) {
                        customerList.add(customer)
                    }
               }
            )
            return customerList
        }

    fun getCustomer(id: Long): Customer? {
        return customerRepository.findById(id).orElse(Customer())
    }

    fun addCustomer(customer: Customer) {
        customerRepository.save(customer)
    }

    fun deleteCustomer(id: Long) {
        customerRepository.deleteById(id)
    }

    fun updateCustomer(customer: Customer, id: Long) {
        val customerInDB = customerRepository.findById(id).get()
        if (customerInDB != null) {
            if (customer.lastName != null) customerInDB.lastName = customer.lastName
            if (customer.firstName != null) customerInDB.firstName = customer.firstName
            if (customer.mail != null) customerInDB.mail = customer.mail
            if (customer.address != null) customerInDB.address = customer.address
            if (customer.phoneNumber != 0) customerInDB.phoneNumber = customer.phoneNumber
            if (customer.age != 0) customerInDB.age = customer.age
            customerRepository.save(customerInDB)
        }
    }
}
